azurerm/resource_arm_key_vault_access_policy.go

```diff
@@ -145,12 +145,6 @@ func resourceArmKeyVaultAccessPolicyCreateOrDelete(d *schema.ResourceData, meta
 
 	keyVault, err := client.Get(ctx, resourceGroup, vaultName)
 	if err != nil {
-		if utils.ResponseWasNotFound(keyVault.Response) {
```

Comment:
We need this check on update, could we change it to:
```
		if utils.ResponseWasNotFound(keyVault.Response) && d.IsNewResource() {
```

Reply:
I think we need this to be `&& !d.IsNewResource()` - I've pushed a replacement commit that fixes makes all of the related acceptance tests pass here and retains the deletion logic.